### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by markelrod** [dynamoose/dynamoose#1243]
### Summary:

Unable to perform query using primary key in 2.8. Works in 2.7.3.

#### Schema
```
// Code here

const BroadcastSchema = new dynamoose.Schema(
  {
    user_id: {
      hashKey: true,
      type: String,
      required: true,
    },
    timestamp: {
      type: Date,
      rangeKey: true,
    },
    session_id: {
      type: String,
      required: true,
      index: {
        name: "session-index",
        global: true,
        throughput: { read: 10, write: 10 },
      },
    },
```

#### Model
```
// Code here
```

#### General
```
// Code here
let broadcasts = await Broadcast.query("user_id")
    .eq(user_id)
    .filter("recording")
    .exists()
    .sort("descending")
    .all(100)
    .exec();
```

### Current output and behavior (including stack trace):

InvalidParameter: Index can't be found for query.
--
at Query.DocumentRetriever.getRequest (/usr/src/app/node_modules/dynamoose/lib/DocumentRetriever.ts:171:11)
at runMicrotasks (<anonymous>)
at processTicksAndRejections (internal/process/task_queues.js:97:5)
at /usr/src/app/node_modules/dynamoose/lib/DocumentRetriever.ts:114:20

### Expected output and behavior:

List of records

### Environment:

Operating System:
Operating System Version:
Node.js version (`node -v`):
NPM version: (`npm -v`):
Dynamoose version:

### Other:
- [X ] I have read through the Dynamoose documentation before posting this issue
- [X ] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [ X] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [X ] I have tested the code provided and am confident it doesn't work as intended
- [ X] I have filled out all fields above
- [X ] I am running the latest version of Dynamoose (HAVE TO REVERT BC OF ISSUE)

**Comment by fishcharlie** [comment]
@markelrod Wanna test the #1240 PR and see if that solves your issue?

**Comment by markelrod** [comment]
Happy to check it. I need to add it to my package.json so Docker build pulls it in. Never tried to reference a PR before. 

I think I use the branch. Is what I need to do?

"dynamoose": "daria-garkusha/ <https://github.com/daria-garkusha/dynamoose/tree/fix-missing-index>dynamoose#fix-missing-inde <https://github.com/daria-garkusha/dynamoose/tree/fix-missing-index>x"

**Comment by fishcharlie** [comment]
@markelrod If you run the following it should work:

```
npm install --save daria-garkusha/dynamoose#fix-missing-index
```

**Comment by markelrod** [comment]
Thanks for confirming. When I use that command it appears to add correctly but after building and deploying I get:

Error: Cannot find module '/usr/src/app/node_modules/dynamoose/dist/index.js'. Please verify that the package.json has a valid "main" entry
api_1             |     at tryPackage (internal/modules/cjs/loader.js:318:19)
api_1             |     at Function.Module._findPath (internal/modules/cjs/loader.js:679:18)
api_1             |     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:949:27)
api_1             |     at Function.Module._load (internal/modules/cjs/loader.js:838:27)
api_1             |     at Module.require (internal/modules/cjs/loader.js:1022:19)
api_1             |     at require (internal/modules/cjs/helpers.js:72:18)
api_1             |     at Object.<anonymous> (/usr/src/app/src/models/User.js:5:1)
api_1             |     at Module._compile (internal/modules/cjs/loader.js:1118:30)
api_1             |     at Module._compile (/usr/src/app/node_modules/pirates/lib/index.js:99:24)
api_1             |     at Module._extensions..js (internal/modules/cjs/loader.js:1138:10) {
api_1             |   code: 'MODULE_NOT_FOUND',
api_1             |   path: '/usr/src/app/node_modules/dynamoose/package.json',
api_1             |   requestPath: 'dynamoose'


Looks like there is not a `dist` directory:


/usr/src/app/node_modules/dynamoose # ls
CHANGELOG.md        CONTRIBUTING.md     README.md           SPONSOR.md          internal            node_modules        package-lock.json   publish             tsconfig.json
CODE_OF_CONDUCT.md  LICENSE             SECURITY.md         docs                lib                 nyc.config.js       package.json        test


Advice? Thanks for walking me through this!

**Comment by fishcharlie** [comment]
@markelrod I hate TypeScript sometimes... I'm not sure how to fix that off the top of my head. You could try going into the `node_modules/dynamoose` folder and run `npm run build`. But not sure that'll work. I thought it should have done that automatically tho. So I'm really not sure.

**Comment by fishcharlie** [comment]
This should be fixed in #1244
